src/include/simeng/MemoryInterface.hh

```diff
@@ -18,7 +18,7 @@ struct MemoryAccessTarget {
   /** The address to access. */
   uint64_t address;
   /** The number of bytes to access at `address`. */
-  uint8_t size;
+  uint16_t size;
```

Comment:
Since this size has been changed from `uint8_t` and bandwidth constraints are already being applied elsewhere, I'm wondering if size threshold should be changed in readBufferThen i.e this statement `uint64_t numBytes = std::min<uint64_t>(length, 128);` given that we are emulating syscalls anyway wouldn't it be better to fetch as much data as possible during a syscall emul?